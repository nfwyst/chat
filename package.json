{
  "name": "chat",
  "version": "1.0.0",
  "main": "./dist/main.js",
  "license": "MIT",
  "scripts": {
    "clean": "rimraf clean",
    "dev": "yarn clean && webpack --watch",
    "start": "electron .",
    "test": "jest"
  },
  "roots": [
    "/Users/cadisson/Work/chat/app"
  ],
  "devDependencies": {
    "babel": "^6.23.0",
    "babel-cli": "^6.26.0",
    "babel-core": "^6.26.3",
    "babel-jest": "^23.4.0",
    "babel-loader": "^7.1.5",
    "babel-plugin-transform-class-properties": "^6.24.1",
    "babel-plugin-transform-object-rest-spread": "^6.26.0",
    "babel-polyfill": "^6.26.0",
    "babel-preset-env": "^1.7.0",
    "babel-preset-es2017": "^6.24.1",
    "babel-preset-react": "^6.24.1",
    "css-loader": "^1.0.0",
    "devtron": "^1.4.0",
    "electron": "^2.0.5",
    "electron-builder": "^20.26.0",
    "electron-debug": "^2.0.0",
    "electron-devtools-installer": "^2.2.4",
    "enzyme": "^3.3.0",
    "enzyme-adapter-react-16": "^1.1.1",
    "file-loader": "^1.1.11",
    "html-webpack-plugin": "^3.2.0",
    "jest": "^23.4.1",
    "photonkit": "^0.1.2",
    "prop-types": "^15.6.2",
    "react": "^16.4.1",
    "react-dom": "^16.4.1",
    "react-test-renderer": "^16.4.1",
    "rimraf": "^2.6.2",
    "style-loader": "^0.21.0",
    "uglifyjs-webpack-plugin": "^1.2.7",
    "webpack": "^4.16.2",
    "webpack-cli": "^3.1.0",
    "webpack-dev-server": "^3.1.5",
    "webpack-merge": "^4.1.3"
  },
  "dependencies": {
    "nodejs-websocket": "^1.7.1",
    "react-timeago": "^4.1.9"
  }
}
