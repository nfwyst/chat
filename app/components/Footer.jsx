import React from "react";
import * as manifest from '../../package.json';

export default class Footer extends React.Component {
  render() {
    return (
      <footer className="toolbar toolbar-footer">
        <h1 className="title">
          { manifest.name } v.{ manifest.version }
        </h1>
      </footer>
    )
  }
}
