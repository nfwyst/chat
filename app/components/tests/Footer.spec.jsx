import * as React from 'react';
import Footer from '../Footer.jsx';
import  * as renderer from 'react-test-renderer';
import * as manifest from '../../../package.json';
import Enzyme, { shallow } from 'enzyme';
import Adapter from "enzyme-adapter-react-16";
Enzyme.configure({ adapter: new Adapter() });

describe('Footer', () => {
  it('match previously saved snaphsot', () => {
    const tree = renderer.create(<Footer />);
    expect(tree.toJSON()).toMatchSnapshot();
  });
  it('render manifest name', () => {
    const tree = shallow(<Footer />);
    expect(tree.find('footer').length).toBe(1);
    expect(tree.find('footer').text().indexOf(manifest.name)).not.toBe(-1);
  });
});
