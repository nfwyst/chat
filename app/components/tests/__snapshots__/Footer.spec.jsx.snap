// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer match previously saved snaphsot 1`] = `
<footer
  className="toolbar toolbar-footer"
>
  <h1
    className="title"
  >
    chat
     v.
    1.0.0
  </h1>
</footer>
`;
